### ISSUE TITLE
Fastboot flash boot failed

### ISSUE THREAD

**Issue by Sp1ke47** [R0rt1z2/mtk-bpf-patcher#2]
Running the script on Linux and got a .gz file. Unziped it to get the .img. When I tried to flash it via fastboot an error occured (remote: 'image is not a boot image')
logs:
```
[?] Detected BOOT_IMAGE for /root/temp/boot.img!
[!] Extracting kernel from /root/temp/boot.img...
[+] Found a match for '48010035822240b980420491e10313aa'!
[!] Compressing the patched kernel binary...
[+] Success. The patched kernel is available at /root/temp/boot1.img.gz!
```
The original kernel and patched file attached. Please help!

[boot_original.zip](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917140/boot_original.zip)
[boot1.img.gz](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917212/boot1.img.gz)

**Comment by R0rt1z2** [comment]
> Running the script on Linux and got a .gz file. Unziped it to get the .img. When I tried to flash it via fastboot an error occured (remote: 'image is not a boot image') logs:
> 
> ```
> [?] Detected BOOT_IMAGE for /root/temp/boot.img!
> [!] Extracting kernel from /root/temp/boot.img...
> [+] Found a match for '48010035822240b980420491e10313aa'!
> [!] Compressing the patched kernel binary...
> [+] Success. The patched kernel is available at /root/temp/boot1.img.gz!
> ```
> 
> The original kernel and patched file attached. Please help!
> 
> [boot_original.zip](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917140/boot_original.zip) [boot1.img.gz](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917212/boot1.img.gz)

I think you're confusing whatever the script is supposed to return you. As it stands currently, it isn't capable of repacking boot images yet. Thus, in your specific situation, it only returns the compressed zImage.

You are supposed to manually repack the resulting `boot1.img.gz` using a third-party tool, such as [Android Image Kitchen (AIK)](https://forum.xda-developers.com/t/tool-android-image-kitchen-unpack-repack-kernel-ramdisk-win-android-linux-mac.2073775/), for example.

**Comment by Sp1ke47** [comment]
True. After repacking the kernel binary into `boot.img` the flash process succeeded. However, when I attempted to patch the newly packed image via Magisk and flashed it, my phone began to bootloop. `vbmeta.img` has been flashed to disable AVB verification. Haven’t tried patching in Magisk first then using this script. Maybe there are some arguments idk between the ramdisk patch and kernel patch.

**Comment by R0rt1z2** [comment]
> True. After repacking the kernel binary into `boot.img` the flash process succeeded. However, when I attempted to patch the newly packed image via Magisk and flashed it, my phone began to bootloop. `vbmeta.img` has been flashed to disable AVB verification. Haven’t tried patching in Magisk first then using this script. Maybe there are some arguments idk between the ramdisk patch and kernel patch.

Could you elaborate that, please? Did the phone boot without the Magisk patch? What kind of bootlop are you experiencing?

**Comment by dtisfg** [comment]
<img width="725" alt="image" src="https://github.com/R0rt1z2/mtk-bpf-patcher/assets/131782042/c89de505-485b-4196-bc41-a4f8aa9ea9ba">
i don't know how to solve this problem, can you help me?

**Comment by LineageOSDefender** [comment]
> Running the script on Linux and got a .gz file. Unziped it to get the .img. When I tried to flash it via fastboot an error occured (remote: 'image is not a boot image')
> logs:
> ```
> [?] Detected BOOT_IMAGE for /root/temp/boot.img!
> [!] Extracting kernel from /root/temp/boot.img...
> [+] Found a match for '48010035822240b980420491e10313aa'!
> [!] Compressing the patched kernel binary...
> [+] Success. The patched kernel is available at /root/temp/boot1.img.gz!
> ```
> The original kernel and patched file attached. Please help!
> 
> [boot_original.zip](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917140/boot_original.zip)
> [boot1.img.gz](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917212/boot1.img.gz)
> 
> 

X2

**Comment by LineageOSDefender** [comment]
> > Running the script on Linux and got a .gz file. Unziped it to get the .img. When I tried to flash it via fastboot an error occured (remote: 'image is not a boot image') logs:
> > 
> > ```
> > [?] Detected BOOT_IMAGE for /root/temp/boot.img!
> > [!] Extracting kernel from /root/temp/boot.img...
> > [+] Found a match for '48010035822240b980420491e10313aa'!
> > [!] Compressing the patched kernel binary...
> > [+] Success. The patched kernel is available at /root/temp/boot1.img.gz!
> > ```
> > 
> > The original kernel and patched file attached. Please help!
> > 
> > [boot_original.zip](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917140/boot_original.zip) [boot1.img.gz](https://github.com/R0rt1z2/mtk-bpf-patcher/files/11917212/boot1.img.gz)
> 
> I think you're confusing whatever the script is supposed to return you. As it stands currently, it isn't capable of repacking boot images yet. Thus, in your specific situation, it only returns the compressed zImage.
> 
> You are supposed to manually repack the resulting `boot1.img.gz` using a third-party tool, such as [Android Image Kitchen (AIK)](https://forum.xda-developers.com/t/tool-android-image-kitchen-unpack-repack-kernel-ramdisk-win-android-linux-mac.2073775/), for example.

I tried, but AIK cannot recognize the output file as image file

**Comment by ghost** [comment]
Just found this and it works great.
Thank you. @R0rt1z2 

@Sp1ke47 @dtisfg @LineageOSDefender 
Here's what I did:

- Grab **boot.img** from your firmware/ROM.
- Then unpack it with **Android Image Kitchen**
- I use the Linux version by executing the **unpackimg.sh** script.
`./unpackimg.sh boot.img`
- After that you will get the **boot.img-kernel** file in the **split_img** folder.

Now it's time to use **mtk-bpf-patcher**

- In my case just do this command to patch **boot.img**:
`mtk-bpf-patcher boot.img boot`
- And the result is the **boot.gz** file.
- Now rename **boot.gz** to **boot.img-kernel**.
- ​​​Then overwrite the **boot.img-kernel** file in the **split_img** folder with the **boot.img-kernel** file you just created.

Go back to **Android image Kitchen** to repack **boot.img**.
- In my case by executing the **repackimg.sh** script without any input.
- Then you will get the **image-new.img** file.
- That's the **patched boot.img**

Tested with GSI.

**Comment by ping-localhost** [comment]
> Just found this and it works great. Thank you. [@R0rt1z2](https://github.com/R0rt1z2)
> 
> [@Sp1ke47](https://github.com/Sp1ke47) [@dtisfg](https://github.com/dtisfg) [@LineageOSDefender](https://github.com/LineageOSDefender) Here's what I did:
> 
> * Grab **boot.img** from your firmware/ROM.
> * Then unpack it with **Android Image Kitchen**
> * I use the Linux version by executing the **unpackimg.sh** script.
>   `./unpackimg.sh boot.img`
> * After that you will get the **boot.img-kernel** file in the **split_img** folder.
> 
> Now it's time to use **mtk-bpf-patcher**
> 
> * In my case just do this command to patch **boot.img**:
>   `mtk-bpf-patcher boot.img boot`
> * And the result is the **boot.gz** file.
> * Now rename **boot.gz** to **boot.img-kernel**.
> * ​​​Then overwrite the **boot.img-kernel** file in the **split_img** folder with the **boot.img-kernel** file you just created.
> 
> Go back to **Android image Kitchen** to repack **boot.img**.
> 
> * In my case by executing the **repackimg.sh** script without any input.
> * Then you will get the **image-new.img** file.
> * That's the **patched boot.img**
> 
> Tested with GSI.

You are amazing. Thank you for this very thorough write-up. Worked for me and an old Android 12 no-name tablet!

**Comment by Ke1i** [comment]
@ping-localhost Is the resulting patched boot image 
supposed to be smaller than the original boot.img?
I've tried but the resulting patched boot.img is a lot smaller than the original. Is this normal?
Attached are some screengrabs.

![Image](https://github.com/user-attachments/assets/490d7d8c-b85d-4ea2-843f-317faecc3314)
![Image](https://github.com/user-attachments/assets/4fc25857-8844-43f1-8502-91731e889af8)
![Image](https://github.com/user-attachments/assets/df1f5d7f-95b2-435b-957f-f685783c3a9c)
![Image](https://github.com/user-attachments/assets/4143aefb-07d7-4c68-8d5d-d6e9e682233e)

**Comment by ping-localhost** [comment]
@Ke1i not sure anymore, sorry. I did use Windows to do the actual steps, so that might have affected the results. No access to my PC today, but will try and look this weekend.

**Comment by Ke1i** [comment]
No problem. I had to use magiskboot-linux for unpacking and repacking which resulted in correct boot.img size.
